src/platforms/hosted/gdb_if.c

```diff
@@ -140,9 +140,15 @@ static sockaddr_storage_s sockaddr_prepare(const uint16_t port)
 		return (sockaddr_storage_s){AF_UNSPEC};
 	}
 
-	/* Pick the first result, copy it and free the list structure getaddrinfo() returns */
+	/* Try to find an IPv6 result but fall back to the first result if none can be found */
 	sockaddr_storage_s service;
 	memcpy(&service, results->ai_addr, family_to_size(results->ai_addr->sa_family));
+	for (addrinfo_s *curr = results; curr; curr = curr->ai_next)
```

Comment:
Because of the inner multi-statement block, the `for` loop is considered to be a multi-statement block too for the style guide, please include the `{}`'s here even though this is technically correct (reduces confusion).